sample/sample/sample.cpp

```diff
@@ -0,0 +1,86 @@
+// sample.cpp : Defines the entry point for the console application.
+//
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+
+using namespace signalr;
+using namespace std;
+using namespace utility;
+
+void runRawConnection()
+{
+	connection * conn = new connection(L"http://localhost:40476/raw-connection");
+
+	conn->set_message_received([conn](const string_t& payload)
+	{
+		cout << "received message:" << endl;
+		wcout << payload << endl;
+
+		if (payload.find(L"stop") != string::npos)
+		{
+			cout << "stop connection" << endl;
+			conn->stop().wait();
+		}
+	});
+
+	conn->start().wait();
```

Comment:
In case of SignalR client it does not matter match but in general you should use `.get()` - AFAIR .wait() won't break on cancellations which, if not handled, will crash your process.